compute_csa.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+#
+# Compute CSA on output segmentations. Input is data_processed_clean
+#
+# Usage:
+#   ./compute_csa.sh <SUBJECT>
+#
+# Manual segmentations or labels should be located under:
+# PATH_DATA/derivatives/labels/SUBJECT/anat/
+#
+# Authors: Sandrine Bédard
+
+# The following global variables are retrieved from the caller sct_run_batch
+# but could be overwritten by uncommenting the lines below:
+# PATH_DATA_PROCESSED="~/data_processed"
+# PATH_RESULTS="~/results"
+# PATH_LOG="~/log"
+# PATH_QC="~/qc"
+
+set -x
+# Immediately exit if error
+set -e -o pipefail
+
+# Exit if user presses CTRL+C (Linux) or CMD+C (OSX)
+trap "echo Caught Keyboard Interrupt within script. Exiting now.; exit" INT
+
+# Retrieve input params
+SUBJECT=$1
+PATH_PRED_SEG=$2
+
+# get starting time:
+start=`date +%s`
+
+# Display useful info for the log, such as SCT version, RAM and CPU cores available
+sct_check_dependencies -short
+
+# Go to folder where data will be copied and processed
+cd ${PATH_DATA_PROCESSED}
+# Copy list of participants in processed data folder
+if [[ ! -f "participants.tsv" ]]; then
+  rsync -avzh $PATH_DATA/participants.tsv .
+fi
+# Copy list of participants in resutls folder
+if [[ ! -f $PATH_RESULTS/"participants.tsv" ]]; then
+  rsync -avzh $PATH_DATA/participants.tsv $PATH_RESULTS/"participants.tsv"
+fi
+# Copy source images
+rsync -avzh $PATH_DATA/$SUBJECT .
+
+# FUNCTIONS
+# ==============================================================================
+
+# Check if manual label already exists. If it does, copy it locally.
+# NOTE: manual disc labels should go from C1-C2 to C7-T1.
+label_if_does_not_exist(){
+  local file="$1"
+  local file_seg="$2"
+  local ofolder="$3"
+  # Update global variable with segmentation file name
+  FILELABEL="${file}_discs"
+  FILELABELMANUAL="${PATH_DATA}/derivatives/labels/${SUBJECT}/${FILELABEL}.nii.gz"
+  
+  echo "Looking for manual label: $FILELABELMANUAL"
+  if [[ -e $FILELABELMANUAL ]]; then
+    echo "Found! Using manual labels."
+    rsync -avzh $FILELABELMANUAL ${FILELABEL}.nii.gz
+    # Generate labeled segmentation from manual disc labels
+    sct_label_vertebrae -i ${file}.nii.gz -s ${file_seg}.nii.gz -discfile ${FILELABEL}.nii.gz -c t2 -ofolder $ofolder
+  else
+    echo "Not found. cannot compute CSA."
+  fi
+}
+
+find_contrast(){
+  local file="$1"
+  local dwi="dwi"
+  if echo "$file" | grep -q "$dwi"; then
+    echo  "./${dwi}/"
+  else
+    echo "./anat/"
+  fi
+}
+
+
+# SCRIPT STARTS HERE
+# ==============================================================================
+# Go to anat folder where all structural data are located
+cd ${SUBJECT}
+
+# Initialize filenames
+file_t1="${SUBJECT}_T1w"
+file_t2="${SUBJECT}_T2w"
+file_t2s="${SUBJECT}_T2star"
+file_t1w="${SUBJECT}_acq-T1w_MTS"
```

Comment:
might need to be updated if we decide to update the data-multi-subject to latest BIDS